WIP: Add a sample utilizing uCAM-III camera module (#14596)
zephyrproject-rtos/zephyr: samples/camera/README.rst

.. _camera-sample:

> Camera Application

Reviewer: Anticipating we may add other camera modules later, this label and title are a bit too generic.  Better would be:
```
.. _ucam-camera-sample:

uCAM-III Camera Application
###########################
```

##################

Overview
********

> This camera application is actually a port of the 4D Systems uCAM-III camera

Reviewer: Let's include a link to the uCAM website:
```suggestion
This camera application is a Zephyr port of the `4D Systems uCAM-III camera`_
```

> module in Zephyr. It defines its own API.

Reviewer: ```suggestion
module. The camera API is defined in :zephyr_file:`samples/camera/src/ucam3.h`.
```

> 

Reviewer: Add:
```
.. _4D Systems uCAM-III camera:
   https://www.4dsystems.com.au/product/uCAM_III/

Requirements
************

The demo assumes that a uCAM-III is connected to the UART_2 TX/RX of a Nucleo
L496ZG. Also, the camera module should be powered by the 5V pin.

Building and Running
********************

> This sample outputs the process of interfacing with the camera module to the

Reviewer: I'm not sure what "outputs the process of interfacing with the camera module" means?  Does it write some debugging output to the console?  (If so can you show an example here.) Does it only produce still images or a video stream? Is there a JPEG image captured somewhere?  What is the API presented by the camera module (link to that would be good)?  Is the whole API implemented in this port?
